synapse/storage/registration.py

```diff
@@ -815,3 +819,34 @@ def get_user_pending_deactivation(self):
             allow_none=True,
             desc="get_users_pending_deactivation",
         )
+
+    @defer.inlineCallbacks
+    def _bg_user_threepids_grandfather(self, progress, batch_size):
+        """We now track which identity servers a user binds their 3PID to, so
+        we need to handle the case of existing bindings where we didn't track
+        this.
+
+        We do this by grandfathering in existing user threepids assuming that
+        they used one of the server configured trusted identity servers.
```

Comment:
one of, or all of? (surely it should be one of, but the code does all of)

Reply:
We assume that they used "one of", but we don't know which one so we have to include all the id servers.